### PR TITLE
Move RpoRandomCoin to crate

### PR DESCRIPTION
**Pull request by Al-Kindi-0**
## Describe your changes
This moves the `RpoRandomCoin` to the `crypto` crate and defines a trait for generating random `Felt` and `Word` on it.

## Checklist before requesting a review
- Repo forked and branch created from `next` according to naming convention.
- Commit messages and codestyle follow [conventions](./CONTRIBUTING.md).
- Relevant issues are linked in the PR description.
- Tests added for new functionality.
- Documentation/comments updated according to changes.

### CODE REVIEW

File: src/rand/mod.rs [hunk 1]
```diff
@@ -1,3 +1,18 @@
...
 pub use winter_crypto::{RandomCoin, RandomCoinError};
+
+use crate::{Felt, Word, ZERO};
+
+mod rpo;
+
+/// Pseudo-random basefield elements generator.
+///
+/// An instance can be used to draw, uniformly at random, basefield elements as well as `Word`s.
+pub trait RandFeltsGen {
```
Comment (bobbinth) [inline src/rand/mod.rs:12]: Not sure if I like `RandFeltsGen` better than `Rng`. Should we just go with `Rng`? Or maybe `FeltRng`?

Also minor nit: on line 9 I'd probably say just: "Pseudo-random element generator."

**Review by bobbinth — APPROVED** [review]
Looks good! Thank you! I left a few small comments inline. After these are addressed, we can merge.

File: src/rand/rpo.rs [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+        self.state[RATE_START] += data[0];
+        self.state[RATE_START + 1] += data[1];
+        self.state[RATE_START + 2] += data[2];
+        self.state[RATE_START + 3] += data[3];
+
+        // Absorb
+        Rpo256::apply_permutation(&mut self.state);
+    }
+
+    fn check_leading_zeros(&self, value: u64) -> u32 {
```
[the remark this answers is not in the document]
Comment (Al-Kindi-0) [reply src/rand/rpo.rs:79]: Are you thinking of adding more than what already exists at the trait level?

File: src/rand/rpo.rs [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+impl RandomCoin for RpoRandomCoin {
+    type BaseField = Felt;
+    type Hasher = Rpo256;
+
+    fn new(seed: &[Self::BaseField]) -> Self {
+        let mut state = [ZERO; STATE_WIDTH];
+
+        let digest = Rpo256::hash_elements(seed);
+        let digest_elem = digest.as_elements();
```
[the remark this answers is not in the document]
Comment (Al-Kindi-0) [reply src/rand/rpo.rs:51]: Makes sense

File: src/rand/mod.rs (hunk 1 above)
Comment (Al-Kindi-0) [reply src/rand/mod.rs:12]: Makes sense! Updated to `FeltRng` and changed the comment.

File: src/rand/rpo.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (bobbinth) [reply src/rand/rpo.rs:77]: Actually, I think that one is sufficient (I just forgot what it was and it wasn't readily available :) ).

**Comment by bobbinth** [comment]
Two other things for this PR:

- Could we add a small section to the main `README` file about the `rand` module?
- Let's update the `CHANGELOG`.

**Review by bobbinth — APPROVED** [review]
All looks good! Thank you!

Should we rebase it from `next` and merge? (this way we don't have to wait until #236 is done.

**Comment by Al-Kindi-0** [comment]
Makes sense. Done!

**Comment by bobbinth** [comment]
Thank you!

**Pull request merged**